pkg/apis/maistra/conversion/extension_providers_test.go

```diff
@@ -172,6 +172,46 @@ meshConfig:
       port: 8000
   - name: prometheus
     prometheus: {}
+`,
+		},
+		{
+			name: "envoyExtAuthzGrpc." + ver,
+			spec: &v2.ControlPlaneSpec{
+				Version: ver,
+				MeshConfig: &v2.MeshConfig{
+					ExtensionProviders: []*v2.ExtensionProviderConfig{
+						{
+							Name: "ext-authz-grpc",
+							EnvoyExtAuthzGRPC: &v2.ExtensionProviderEnvoyExternalAuthorizationGRPCConfig{
+								Service:       "ext-authz.foo.svc.cluster.local",
+								Port:          8000,
+								Timeout:       strPtr("1s"),
+								FailOpen:      true,
+								StatusOnError: "500",
+								IncludeRequestBodyInCheck: &v2.ExtensionProviderEnvoyExternalAuthorizationRequestBodyConfig{
+									MaxRequestBytes:     int64Ptr(100),
+									AllowPartialMessage: boolPtr(true),
+									PackAsBytes:         boolPtr(true),
+								},
+							},
+						},
+					},
+				},
+			},
+			helmValues: `
+meshConfig:
+  extensionProviders:
+  - name: ext-authz-grpc
+    envoyExtAuthzGrpc:
+      service: ext-authz.foo.svc.cluster.local
+      port: 8000
+      timeout: 1s
+      failOpen: true
+      statusOnError: "500"
+      includeRequestBodyInCheck:
+        maxRequestBytes: 100
+        allowPartialMessage: true
+        packAsBytes: true
```

Comment:
Are these field names correct? Is it `failOpen` or `fail_open`? Both formats appear in the upstream type, but I'm not sure which one is actually used:
```
FailOpen bool `protobuf:"varint,4,opt,name=fail_open,json=failOpen,proto3" json:"fail_open,omitempty"`
                                           ^^^^^^^^^      ^^^^^^^^               ^^^^^^^^^
```

Reply:
We were following Jacek's work by doing the camel casing instead of snake casing. If we want to be more inline with upstream, I am more than happy to change! :)

Reply:
Yes, I know, but I'm talking specifically about the helmValues, not about the SMCP (where, as you say, we followed Jacek's work). For helmValues, we need to use whatever istio expects.

Reply:
Okay, it looks like it expects the fields to be in camel case, so we're good.

https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig-ExtensionProvider-EnvoyExternalAuthorizationHttpProvider

Reply:
@luksa we use camel case, because we only support JSON schema.